### DISCUSSION TITLE
lolMiner 1.25

### DISCUSSION THREAD

**Discussion by Lolliedieb** [Lolliedieb/lolMiner-releases#421] [General]
# lolMiner 1.25

### lolMiner goes CUDA!
Added a real Cuda back-end for better Nvidia GPU support on Ethash. Features:
- Supports Maxwell to Ampere GPU generations.
- Two different mining kernels. Use --mode a (faster) --mode b (better energy efficiency) to select between the two. The selection can be done per card via a comma separated list. In mixed system select 'a' for skipping over the AMD cards. 
- Both kernel modes need less energy and perform better then in 1.24a
- Reduced internal latency for less stale shares
- lolMiner works now without OpenCL driver installed
- In case of mixed rigs AMD GPUs will use OpenCL while Nvidia cards use Cuda
- ZIL cache feature fully supported (and stable)
- Temperature stop & Zombie mode is currently not supported

### Further Changes (over 1.24a)
- Added Ethash, Beam Hash III, Grin Cuckatoo 32 and Cortex kernels for RX 6700
- The Ethash stratum interface will now try to run up to three attempts of reconnecting before switching the stratum mode

### Bug fixes
- Fixed "Warning: index out of bounds" error when switching from ETHPROXY to ETHV1 stratum mode. This might solve problems with some pools on connection loss. 

### Recommendations for Cuda backend
- Recommendation: When using lolMiner on Nvidia cards only use "--watchdog exit" mode and run the miner in a script that will automatically restart it on closing. 
- For ideal efficiency fix the core clock, do not use the offset functionality. Recommended values for selected GPUs:

| GPU        | Range       | 
| ------------- |:-------------:| 
| 2070 | 1000 - 1050 |
| 2080 | 1110 - 1160 |
| 3060 (1) | 1070 - 1120 |
| 3060ti | 1300 - 1350 |
| 3070 | 750 - 800 |
| 3080 | 1010 - 1060 |

(1) Using Windows and Nvidia Driver 470.05 Beta

<hr /><em>This discussion was created from the release <a href='https://github.com/Lolliedieb/lolMiner-releases/releases/tag/1.25'>lolMiner 1.25</a>.</em>

**Comment by shredz88** [comment]
**Avoid 1.25!!! Ended up with a trojan...**

**Reply by Lolliedieb** [reply]
A trojan?
Certainly not ... I even signed the executable for Windows so I prove it is as compiled here. 

By the way here is the virustotal reports. 
Windows:
https://www.virustotal.com/gui/file/ba7470920e0ccce2763e81086c3d562a279d476462fa3113b3e9d67b1d014084/detection

Linux:
https://www.virustotal.com/gui/file/b7e57d685c9bc253325c5bfe908604d91a6b68480a736ad72fd66370f7c7d5b5/detection

As you can see it is often reported as some kind of "CoinMiner" ... well ... thats what it is supposed to be, right? Most engines react by the way because of OpenCL / Cuda in it + use of an executable packer that makes lolMiner much smaller then it would be without (4 vs about 30 MByte)...

**Reply by shredz88** [reply]
I have used previous lolminer deployments as long as I have been cryptomining without issue (and I thank you for your work). Although, I cannot rebuttle anything that you claim is true; I am simply not knowlegble enough in how the programming works, but here is my individual experience with using version 1.25:
- (/) allowing any .exe to run on your PC over a network, its technically a "virus", but it is a necessary part of crypto mining.
- (/) developers such as lolminer, take a fee % for supplying the .exe algorithm(s), buts it is a necessary part of cryptomining.
- (/) previous versions of lolminer ran with coinminer (riskware) but with no affect on PC performance or anything suspiciously malicious.

- (PRO) 1.25 - ran more efficiently and boosted my hash rate by 3-5%
- (PRO) 1.25 - GPUs ran a few degrees cooler.
- (PRO) 1.25 - GPUs appeared to be highly stable. (0 crashes)

- (CON) 1.25 - antivirus software detected 4 Trojan.coinminers on each of my mining rigs.
- (CON) 1.25 - the lolminer CMD window was hijacked, and was displaying an astronomical hashrate (1XXXXXXX.XX mh/s) to an foreign account address to mine.
- (CON) 1.25 - My miner resources were being drawn from maliciously and appeared to be operating in reduced load "stealth mode". (but I could see the .exe running)
- (CON) 1.25 - Basic windows functions were disabled and had to hard restart.

*****I personally recommend not using lolminer 1.25 based on my experience, but users can use their own discretion on whether or not they choose to use version 1.25*****

**Reply by Lolliedieb** [reply]
The last cons: what cards were this? Could be the cuda cards did crash on startup and did not clearly shut down... in fact then they did nothing useful just showing a high hr (which in fact was just
kernel start
kernel crash
miner detects kernel is done and adds processed nonces to statistics
repeat
which means stats are off while card did ... nothing

**Reply by Lolliedieb** [reply]
By the way: when you do track the lolMiner network traffic you will see exactly 2 connections: one to your pool and one for the fee pool on a well known mining pool - there is no other connection done by the miner. Also on many systems the miner can run on a write protected file. You will see it neither writes nor reads nor modifies any file on your system :)

**Reply by shredz88** [reply]
I'm using 3090s. Like I said, I have had no issues with the past versions, but 1.25 is giving me red flags.  So, personally I'm going to stay with the previous version (1.21); even if it's for my own psuedo-safety. (I do appreciate the quick replies)

**Reply by adamya-singh** [reply]
> A trojan?
> Certainly not ... I even signed the executable for Windows so I prove it is as compiled here.
> 
> By the way here is the virustotal reports.
> Windows:
> https://www.virustotal.com/gui/file/ba7470920e0ccce2763e81086c3d562a279d476462fa3113b3e9d67b1d014084/detection
> 
> Linux:
> https://www.virustotal.com/gui/file/b7e57d685c9bc253325c5bfe908604d91a6b68480a736ad72fd66370f7c7d5b5/detection
> 
> As you can see it is often reported as some kind of "CoinMiner" ... well ... thats what it is supposed to be, right? Most engines react by the way because of OpenCL / Cuda in it + use of an executable packer that makes lolMiner much smaller then it would be without (4 vs about 30 MByte)...

It also reported "Program:Win32/Uwamson.A!ml" for me, which looks like a really bad virus after a quick google search. Can you tell me if this is a false positive or should I take it seriously?

**Reply by jgonzis** [reply]
Nearly all miners said that they are virus. This is quite common...

**Reply by howelljs** [reply]
Hello everyone,

Norton 360 blocked lolminer yesterday (4/20/2021) flagging it with risk name: Trojan.Gen.MBT cateogry: Virus risk level: High.

Not sure why after installing and running lolminer 1.25 that it took more than a week for Norton to flag this lolminer as a virus.  However, I think Norton 360 just happen to pick it up on yesterday's scan.  Once Norton 360 quarantined the program, it was devilish and elusive to find all the blocks that would allow lolminer to run.

After a bit of research, the resolution was to open the Norton 360 app and then open Device Security, then click Settings, then click Firewall, then click Intrusion and Browser Protection, then in the Intrusion AutoBlock section click Configure, allow remote computer, then in the Intrusion Signatures section, allow for system infected miner.bitcoinminer activity 16, System Infected: Bitcoinminer Activity 7, etc., until it started running again.

TL;DR - I'm still not sure if this is a real threat or false positive.  Either way allowing lolminer to run, I'm back mining crypto. 

![lolminer trojan gen mbt](https://user-images.githubusercontent.com/6771265/115549084-3c832380-a276-11eb-9f77-040f8ce431c0.jpg)

![lolminer norton firewall intrusion and browser protection](https://user-images.githubusercontent.com/6771265/115549284-76ecc080-a276-11eb-8573-bb46f6a806f3.JPG)

**Reply by jgonzis** [reply]
It is a signed executable... be sure you download from oficial repository Github...

**Reply by howelljs** [reply]
I did download from here ... so not sure what went down.

**Comment by Gibswal** [comment]
![Capture](https://user-images.githubusercontent.com/82226070/114201544-3758d700-994e-11eb-95a1-f2e6b5912777.PNG)
How do I do this? Sorry if noob question I have only been mining for a couple of weeks now. Thanks!

**Reply by Lolliedieb** [reply]
These are just options to add in command line
Append 
--mode a 
to the command in .bat or .sh that starts the miner  and you will switch to the alternative kernel.

**Reply by Gibswal** [reply]
> These are just options to add in command line
> Append
> --mode a
> to the command in .bat or .sh that starts the miner and you will switch to the alternative kernel.

Thanks for the reply, I don't think I have managed to get it working though. Where exactly do I need to put "--mode a" ?
I am using mine_eth.bat
Thanks again for the help.

**Reply by Reinsettle** [reply]
@echo off
setlocal enableDelayedExpansion

set "POOL=us1.ethermine.org:4444"
set "WALLET=YourAddress.Worker"										

:loop
lolMiner.exe --algo ETHASH --mode a --pool !POOL! --user !WALLET! --watchdog exit
if %ERRORLEVEL% == 42 (
	timeout 10
	goto loop
)
pause

**Reply by cryp-tog** [reply]
I tried something to that effect, the problem is Windows is still hanging onto the executable and not getting to the next line.  So whether it's goto start or an errorlevel trap, it won't work.  My buddy did figure it out, I don't have his code, but he basically created a parent script to both call lolminer and a choice /t that times out and runs taskkill after an hour or so.  He also just updated to 1.26, not sure if either will actually work, but we'll see.

**Comment by cryp-tog** [comment]
I'm helping my friends mine stuff, and they're finding that the command line buffer (not the window/scroll buffer) for setlocal enableDelayedExpansion runs out every ~10 hours or so, which closes the script, the last line of the batch file run is an error stating this.  So I simply added:

```
:start
all the rest of the stuff
goto start
pause
```

And everything loops and does great.  Really love the improvements in efficiency across both AMD and Nvidia cards; my friends are getting the same hash rates but using a lot less power.  They have not experienced any issues with AV/viruses/anything either, for what it's worth, not discrediting the above remarks.

Additionally, I have one friend with a single 2070 Super in a single machine, and every few hours they get a Windows error stating, "lolMiner.exe has stopped working.  A problem caused the program to stop working correctly.  Windows will close the program and notify you if a solution is available."  With a "Close Program" button as the only option.  When they click on Close Program, the batch script recycles and kicks off the miner again, it runs for a half a day and crashes again.  I think they're mining BEAM.  Any ideas on how to capture or redirect that error?

**Reply by jgonzis** [reply]
I think Beam is not yet implemented with CUDA... that's probably the reason of some crashes

**Reply by cryp-tog** [reply]
Interesting.  Do you think changing mode would fix it?  [--mode a] vs [--mode b]?

I have different friends running 1660's and 1050TI's, they all just upgraded to 1.25 running in --mode a, and they are not having the problems.  It's just this one friend with one 2070 Super in one box.  Ideas?  I have had my buddy alter the script to start lolminer.exe 2>nul, we'll see if that does anything, but I doubt it.  Waiting for it to crash again.

**Reply by jgonzis** [reply]
With a 1660 I suppose they are mining ETH... they should update to 1.25... the 1050 is not able to mine ETH, so unless they are mining ETC they will not see the improvement

**Reply by cryp-tog** [reply]
Everyone is mining BEAM.

**Comment by Reinsettle** [comment]
So I have run it for 3 days now on a single 1060 SSC using the same mem clock settings (0 core) I was using with V1.24 and I'm getting about 3 shares less on average per hour at the pool. It's getting lower maximum shares at 29 from 33, and lower lows at 15 from 19. Not sure why because I'm getting less stale shares from the lower latency. Although not by much

**Reply by cryp-tog** [reply]
Are you running --mode a or --mode b, or have you specified?  I'm a bystander who likely won't have an answer for you, but I'm happy to try to help the effort.  lolMiner is the best.

**Reply by Reinsettle** [reply]
I never specified it. It's a little deceiving since it states *End of user-editable part*

**Reply by Reinsettle** [reply]
Well I just tried -- mode a and it does appear to be (faster)  Facepalm Emoji. Your help is appreciated! Bystander

**Reply by cryp-tog** [reply]
Awesome.  So you can get around the trouble with delayed expansion (if you have it) by doing this instead as well:

`:start
lolMiner.exe --coin BEAM --watchdog exit --mode a --pool beam-us.leafpool.com:3333 --user WALLETADDRESS.WORKERNAME
goto start`

This is the whole batch script, you don't need anything else, it's all filler.  I will be back in like 10 hours or so.  HODL!!!

**Reply by jgonzis** [reply]
:start

lolMiner.exe --coin BEAM --watchdog exit --mode a --pool beam-us.leafpool.com:3333 --user WALLETADDRESS.WORKERNAME

goto start

**Reply by Reinsettle** [reply]
--algo instead of --coin works.

Update: It does have a higher hashrate than 1.24 with --mode a  ; but it is running at 2c warmer and using about 8 watts more energy if that is a concern to some people. I personally have adequate cooling. And it's still cool outside. The summer might be a different story.

**Reply by Reinsettle** [reply]
Well folks I have let It run in --mode a over night and it is performing very well. Highest stable hashrate I have been able to achieve ever with this version and not even 1 stale share overnight. I just wanted to stop in and thank loliedieb for all of the hard work on this. You are changing peoples lives, and a lot of kids lives for the better. You are one of those unseen heroes that sort of changed the world that no one ever hears about. Like a typical scientist

**Reply by Reinsettle** [reply]
I let it run undisturbed for exactly 24 hours with the card absolutely on the edge of crashing(+8 more on mem clock is instant crash just to give you an idea of how on the edge) and for 1, it never did crash, and 2...  3 stale shares in 24 hours.   Keep in mind, I'm just testing all of this on a single 1060 ssc. It held a steady hashrate of 24.23 the entire time. In other words, this is the best miner that exists for ddr6. I haven't gotten my hands on a ddr6x yet :|

**Comment by SpuriousECG** [comment]
Awesome work!

**Comment by hub200** [comment]
Hello,
 after edited  mine_eth and run it. it show me the flowing ,so my question why it failed is something wrong with the gpu
Setup Miner...
OpenCL driver detected.
Number of OpenCL supported GPUs: 1
No Cuda driver or GPUs detected.
Device 0:
    Name:    AMD Radeon(TM) R5 Graphics
    Address: 0:1
    Vendor:  Advanced Micro Devices (AMD)
    Drivers: OpenCL
    Memory:  2559 MByte (2992 MByte free)
    Active:  false (Unsupported device or driver version.)

Connecting to pool...
Connected to us1.ethermine.org:4444  (TLS disabled)
Set Ethash stratum mode: Ethereum Proxy
Authorized worker: 0x1946Af98cd7a45Fc41325Ee034b4D77A17E87E68.lolMinerWorker --watchdog exit
New job received: 0xa12c46 Epoch: 407 Target: 0000000112e0be82
Start Mining...
All devices deselected or failed compatiblity check. Closing lolMinerPress any key to continue . . .

**Reply by cryp-tog** [reply]
You don't have enough RAM to mine ETH.  The ETH DAG size (what your card needs to hold to creates hashes) is over 4Gb now, so a 5Gb card or larger is required.  Here's the rough schedule (it's not an exact science) for all the DAGs out there: https://minerstat.com/dag-size-calculator 

A 3Gb card isn't going to be able to mine much, if anything, unfortunately.  Sorry to bring you bad news.  Try mining a few different things, let the folks here know if you find something you can mine?  It looks like ZIL has a small DAG and isn't worthless, try that?

**Reply by hub200** [reply]
I have also try it on different laptop with 16GB and it say job received but active still false so is that mean 16gb  still not enough ?

Setup Miner...
OpenCL driver detected.
Number of OpenCL supported GPUs: 1
No Cuda driver or GPUs detected.
Device 0:
    Name:    AMD Radeon R7 Graphics
    Address: 0:1
    Vendor:  Advanced Micro Devices (AMD)
    Drivers: OpenCL
    Memory:  12397 MByte (16279 MByte free)
    Active:  false (Unsupported device or driver version.)

Connecting to pool...
Connected to us1.ethermine.org:4444  (TLS disabled)
Set Ethash stratum mode: Ethereum Proxy
Authorized worker: 0x1946Af98cd7a45Fc41325Ee034b4D77A17E87E68.lolMinerWorker --watchdog exit
New job received: 0x49818d Epoch: 407 Target: 0000000112e0be82
Start Mining...
All devices deselected or failed compatiblity check. Closing lolMinerPress any key to continue . . .

**Reply by cryp-tog** [reply]
Then it's got to be the driver?  I'm not sure.  I'm also neither the dev, nor an expert, just happy to try to help.  I see it's identifying as an AMD Radeon R7 with 16Gb RAM, but I'm not aware of an R7 with 16Gb RAM.  That's odd to me.  The only 16Gb cards I can find documented for laptops are Nvidia cards, though again I'll admit I'm not the expert, I can see it's fairly plain on the screen.  LOL.  Try updating the driver?  https://www.amd.com/en/support has a tool for it.

At the same time- I strongly, strongly DO NOT recommend mining on a laptop.  Laptops typically cannot vent heat fast enough, you're very likely going to kill your laptop after a few hours of hard mining.

**Reply by jgonzis** [reply]
You cannot mine with an R7 card... it is too old...

**Comment by ghost** [comment]
![B38A4F53-EE30-411E-9358-1B0B80D9307C](https://user-images.githubusercontent.com/79080183/114852638-3e1a9a80-9de3-11eb-9960-38a035a35b1e.jpeg)
Someone please help with this error

**Reply by jgonzis** [reply]
That happened to me too, but with the loop started next time perfectly

**Reply by ghost** [reply]
What makes that happen? Mine keeps doing that. When using 2 gpu’s it works fine. When a 3rd is added it does that

**Reply by jgonzis** [reply]
> What makes that happen? Mine keeps doing that. When using 2 gpu’s it works fine. When a 3rd is added it does that

Could you try... to run 2 instances?... 

Try the bat file like this...  I suppose you are in 1.26... 

That will run 2 instances one with device 2 that crashed... (it could be also too much OC) and another with device 0,1

start lolMiner.exe --algo ETHASH --pool !POOL! --user !WALLET! --watchdog exit --devices 2
:loop
lolMiner.exe --algo ETHASH --pool !POOL! --user !WALLET! --watchdog exit --devices 0,1
if %ERRORLEVEL% == 42 (
	timeout 10
	goto loop
)

pause

**Reply by ghost** [reply]
Using 1.25 will try 1.26 now. I also thought it might be OC so i closed afterburner and ran on default but same thing.  Will also try 2 instances thanks

**Reply by jgonzis** [reply]
> 
> 
> ![B38A4F53-EE30-411E-9358-1B0B80D9307C](https://user-images.githubusercontent.com/79080183/114852638-3e1a9a80-9de3-11eb-9960-38a035a35b1e.jpeg)
> Someone please help with this error

I know what problem you have... you should increase the swap drive in Windows.

[https://www.pcgamer.com/what-are-page-and-swap-files-and-how-big-should-they-be/](https://www.pcgamer.com/what-are-page-and-swap-files-and-how-big-should-they-be/)

Try custom size and put Initial Size 60000-Maximum Size 60000 .That worked perfectly more me

**Comment by Lolliedieb** [comment]
Afaik it is a false positive.
Just to point out: when you wireshark the miner you see two connections,
yours and fee, not more.
Also it never asks for admin rights and does not write anything to file
system when running in a save box. More I can't do...

XxsuperuserxX ***@***.***> schrieb am Fr., 16. Apr. 2021,
02:16:
